packages/connectivity/src/scp-cf/destination/destination-from-service.ts

```diff
@@ -315,7 +348,7 @@ class DestinationFromServiceRetriever {
     const exchangeTenant = this.getExchangeTenant(destination);
     const clientGrant = await serviceToken('destination', {
       jwt:
-        origin === 'subscriber'
+        origin === 'subscriber' && this.subscriberToken?.type !== 'custom'
```

Comment:
Is this covered by some tests from a test driven perspective?

Reply:
yes, I inverted the logic so I ask for origin `provider` which covers the custom case as well